meson.build

```diff
@@ -185,37 +185,48 @@ endif
 
 # CoreText
 coretext = false
-if host_system == 'darwin' and not get_option('coretext').disabled()
+if host_system == 'darwin'
     coretext_dep = dependency(
         'appleframeworks',
         modules: ['CoreText', 'CoreFoundation'],
-        required: false,
+        required: get_option('coretext'),
     )
```

Comment:
I suspect the `required: get_option('coretext')` addition will prevent the second check for old-style includes from ever running. You can probably try by reversing the check order again.

I’m not familiar enough with CoreText or Apple  frameworks in general to say whether checking the new version first is a good idea or problematic. Any thoughts @astiob?

Reply:
> I think directly linking to CoreText.framework when it is available is the preferable option.

I’m not 100% sure, but I think this may have precisely been the reason to check the _old_ version first: that way, you can build a binary on a newer macOS and run it on an older one, whereas if you link to CoreText.framework eagerly, then the binary will fail to load on an older macOS.